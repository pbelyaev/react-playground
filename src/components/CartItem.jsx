import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CartItemSection from '../styled/CartItem';

class CartItem extends Component {
  render() {
    const totalPrice = this.props.price * this.props.quantity;

    return (
      <CartItemSection>
          { totalPrice }
      </CartItemSection>
    );
  }
}

CartItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
};

export default CartItem;