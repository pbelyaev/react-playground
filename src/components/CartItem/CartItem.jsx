// @flow
import React from "react";
import { CartItemWrapper } from "./helpers/CartItemWrapper";
import { CartItemBlock } from "./helpers/CartItemBlock";

type CartItemProps = {
  id: number,
  name: string,
  price: number,
  quantity: number,
  onRemove: Function
};

export function CartItem(props: CartItemProps) {
  const { price, quantity, name, id, onRemove } = props;
  const totalPrice = price * quantity;

  return (
    <CartItemWrapper>
      <CartItemBlock>{name}</CartItemBlock>
      <CartItemBlock>{`$${price}`}</CartItemBlock>
      <CartItemBlock>{`x${quantity}`}</CartItemBlock>
      <CartItemBlock>{`$${totalPrice}`}</CartItemBlock>
      <CartItemBlock>
        <button
          type="button"
          onClick={() => onRemove(id)}
        >
          Remove
        </button>
      </CartItemBlock>
    </CartItemWrapper>
  );
}
