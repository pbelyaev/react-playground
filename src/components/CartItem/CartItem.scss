.cart-item {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  width: 100%;
  padding: 10px;
  box-sizing: border-box;

  &__block {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
  }

  &__remove-button {
    // ...
  }
}