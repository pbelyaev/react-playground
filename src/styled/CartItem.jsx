import styled from 'styled-components';

export default styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: no-wrap;
  width: 100%;
  padding: 10px;
`;