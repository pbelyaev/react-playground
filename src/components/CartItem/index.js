import CartItem from "./CartItem";
import "./CartItem.scss";

export default CartItem;