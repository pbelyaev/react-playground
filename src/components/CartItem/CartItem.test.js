// @flow
/* eslint-disable */
import React from "react";
import { shallow } from "enzyme";
import {CartItem} from "./CartItem";

describe("CartItem Test Suite", () => {
  const onRemoveMock = jest.fn();
  const wrapper = shallow(
    <CartItem id={1} name="iPhone X" price={999.99} quantity={2} onRemove={onRemoveMock} />
  );

  it("matches the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders name", () => {
    expect(
      wrapper
        .find("CartItemBlock")
        .at(0)
        .render()
        .text()
    ).toBe("iPhone X");
  });

  it("renders price", () => {
    expect(
      wrapper
        .find("CartItemBlock")
        .at(1)
        .render()
        .text()
    ).toBe("$999.99");
  });

  it("renders quantity", () => {
    expect(
      wrapper
        .find("CartItemBlock")
        .at(2)
        .render()
        .text()
    ).toBe("x2");
  });

  it("renders total price", () => {
    expect(
      wrapper
        .find("CartItemBlock")
        .at(3)
        .render()
        .text()
    ).toBe("$1999.98");
  });

  it("calls `onRemove` callback on remove button click", () => {
    wrapper.find('button').simulate('click');
    expect(onRemoveMock).toBeCalled();
    expect(onRemoveMock).toBeCalledWith(1);
  });
});
