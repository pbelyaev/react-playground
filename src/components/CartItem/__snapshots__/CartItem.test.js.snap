// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Test Suite matches the snapshot 1`] = `"<div class=\\"CartItemWrapper-kMZqWe kUcNLL\\"><div class=\\"CartItemBlock-itbsVk jNsRUN\\">iPhone X</div><div class=\\"CartItemBlock-itbsVk jNsRUN\\">$999.99</div><div class=\\"CartItemBlock-itbsVk jNsRUN\\">x2</div><div class=\\"CartItemBlock-itbsVk jNsRUN\\">$1999.98</div><div class=\\"CartItemBlock-itbsVk jNsRUN\\"><button type=\\"button\\">Remove</button></div></div>"`;
